### PR TITLE
Aws docker examples

### PR DESCRIPTION
**Pull request by OndroMih**
Based on the Payara Docker in AWS guide that I worked on.

### CODE REVIEW

File: cloud-providers/aws/beanstalk/aws-eb-docker-remote-deployment/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Deploy an application on AWS Elastic Beanstalk using a Docker image from a Docker repository
+
+AWS Elastic Beanstalk (EB) allows deploying applications using an existing Docker image in a Docker repository. EB will need a `Dockerrun.aws.json` configuration file to deploy and run the application in one or more Docker containers. This JSON file specifies the name of the Docker image to use and overrides the default configuration of that image. EB will automatically scale and create/drop Docker container as needed. EB also automatically creates a load-balancer in front of the containers.
```
Comment (rdebusscher) [inline cloud-providers/aws/beanstalk/aws-eb-docker-remote-deployment/README.md]: Typo

>EB will automatically scale and create/drop Docker container as needed.

EB will automatically scale and create/drop Docker container**s** as needed.

File: cloud-providers/aws/beanstalk/aws-eb-docker-remote-deployment/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+The project needs to contain a `Dockerrun.aws.json` file and all the resources needed to build the Docker image. You can start with this directory which contains a sample `Dockerrun.aws.json` file which will deploy an empty Payara Server using un unmodified Payara Server Docker image downloaded from the public Docker Hub registry.
```
Comment (rdebusscher) [inline cloud-providers/aws/beanstalk/aws-eb-docker-remote-deployment/README.md]: Typo

> using un unmodified Payara Server Docker image

using an unmodified Payara Server Docker image.

File: cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/Dockerfile-Payara-Micro [hunk 3]
```diff
@@ -0,0 +1,7 @@
+FROM payara/micro
+
+COPY target/application.war $DEPLOY_DIR
+
+EXPOSE 8080
```
Comment (rdebusscher) [inline cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/Dockerfile-Payara-Micro]: Expose already done in _payara/micro_ Docker Image, can be removed.

File: cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/Dockerfile-Payara-Micro [hunk 4]
```diff
@@ -0,0 +1,7 @@
+FROM payara/micro
+
+COPY target/application.war $DEPLOY_DIR
```
Comment (rdebusscher) [inline cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/Dockerfile-Payara-Micro]: Put COPY as last command in file for better reuse of Docker layers.

File: cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,7 @@
+FROM payara/server-full
+
+COPY target/application.war $DEPLOY_DIR
+
+ENV DEPLOY_PROPS=--contextroot=/
+
+EXPOSE 8080
```
Comment (rdebusscher) [inline cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/Dockerfile]: Expose already done in _payara/server-full_ Docker Image, can be removed.

File: cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,7 @@
+FROM payara/server-full
+
+COPY target/application.war $DEPLOY_DIR
+
+ENV DEPLOY_PROPS=--contextroot=/
```
Comment (rdebusscher) [inline cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/Dockerfile]: Put ENV before COPY for better Docker Layer reuse.

File: cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/README.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
+# Deploy an application on AWS Elastic Beanstalk using Dockerfile
+
+
+AWS Elastic Beanstalk (EB) allows deploying applications using a Dockerfile. EB will use the Dockerfile to build a Docker image and use it to deploy and run the application in one or more Docker containers. EB will automatically scale and create/drop Docker container as needed. EB also automatically creates a load-balancer in front of the containers.
```
Comment (rdebusscher) [inline cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/README.md]: Typo

> scale and create/drop Docker container as needed

scale and create/drop Docker container**s** as needed

File: cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/pom.xml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+    <groupId>fish.payara.hello</groupId>
+    <artifactId>hello-from-payara</artifactId>
+    <version>1.0</version>
+    <packaging>war</packaging>
+
+    <name>hello-from-payara</name>
+
+    <properties>
+        <endorsed.dir>${project.build.directory}/endorsed</endorsed.dir>
```
Comment (rdebusscher) [inline cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/pom.xml]: Not needed, can be removed

File: cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/pom.xml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <finalName>application</finalName>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
```
Comment (rdebusscher) [inline cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/pom.xml]: No Java Sources, can be removed.

File: cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/Dockerfile (hunk 5 above)
Comment (OndroMih) [reply cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/Dockerfile]: I'd rather leave it here to stress that the port needs to be exposed for EB. It's redundant I know, but it's not documented that the Payara Docker image exposes 8080.

File: cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/Dockerfile-Payara-Micro (hunk 3 above)
Comment (OndroMih) [reply cloud-providers/aws/beanstalk/payara-server-docker-aws-eb/Dockerfile-Payara-Micro]: Same comment as for Payara Server - I'd rather leave it as exposing 8080 by Payara Micro image is not documented.

**Comment by OndroMih** [comment]
Hi @rdebusscher, can you review again / approve? I've update the PR according to your comments, I only think that leaving explicit EXPOSE 8080 in Dockerfile adds clarity so I left it there. I also have it in my guide and video so I'd rather leave it consistent.

**Pull request merged**